pkg/kubelet/prober/worker.go

```diff
@@ -151,6 +160,9 @@ func (w *worker) run() {
 		ProberResults.Delete(w.proberResultsSuccessfulMetricLabels)
 		ProberResults.Delete(w.proberResultsFailedMetricLabels)
 		ProberResults.Delete(w.proberResultsUnknownMetricLabels)
+		ProberDuration.Delete(w.proberDurationSuccessfulMetricLabels)
+		ProberDuration.Delete(w.proberDurationFailedMetricLabels)
```

Comment:
Duration metrics are usually only meaningful when measuring successes since failures will trigger different code paths that are not really useful to measure since the request is bound to fail.
A concrete example is to measure SLOs. In that case, you are interested in the percentage of requests that failed and the percentage of requests that are too slow. You'll have the number of failures with the counter, in your case `prober_probe_total` and the number of slow requests via the histogram `prober_probe_duration`. So you'll never need the latency of the failures.

Reply:
Thanks! will remove